Cell 1 — Mount Drive

In [1]:
# 1) Mount Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Cell 2 — Settings (edit these)

In [2]:
# 2) Settings
PROJECT_DIR   = "/content/drive/MyDrive/WorstellWitnessWaybackMachine"  # <- your repo folder on Drive
SITE_URL      = "https://BillWorstell.github.io/WorstellWitnessWaybackMachine/"  # <- your Pages URL

# PAT is only needed if your remote is a clean URL (no token). A classic PAT with 'repo' scope works.
GITHUB_USERNAME = "BillWorstell"
GH_TOKEN        = "YourPAT"   # <-- paste PAT (or leave blank if your remote already includes a token)

REMOTE_NAME = "origin"  # usually 'origin'
print("Project:", PROJECT_DIR)
print("Site URL:", SITE_URL)

Project: /content/drive/MyDrive/WorstellWitnessWaybackMachine
Site URL: https://BillWorstell.github.io/WorstellWitnessWaybackMachine/


Cell 3 — Install MkDocs and build once (quick sanity check)

In [3]:
# 3) Install & build locally
!python -m pip install -q mkdocs mkdocs-material
%cd $PROJECT_DIR
!mkdocs build -q

print("Local build complete →", f"{PROJECT_DIR}/site/index.html")

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 39.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.2/9.2 MB 96.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 397.9/397.9 kB 26.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 266.2/266.2 kB 18.7 MB/s eta 0:00:00
/content/drive/MyDrive/WorstellWitnessWaybackMachine
/usr/local/lib/python3.12/dist-packages/jieba/__init__.py:44: SyntaxWarning: invalid escape sequence '\.'
  re_han_default = re.compile("([\u4E00-\u9FD5a-zA-Z0-9+#&\._%\-]+)", re.U)
/usr/local/lib/python3.12/dist-packages/jieba/__init__.py:46: SyntaxWarning: invalid escape sequence '\s'
  re_skip_default = re.compile("(\r\n|\s)", re.U)
/usr/local/lib/python3.12/dist-packages/jieba/finalseg/__init__.py:78: SyntaxWarning: invalid escape sequence '\.'
  re_skip = re.compile("([a-zA-Z0-9]+(?:\.\d+)?%?)")
Local build complete → /content/drive/MyDrive/WorstellWitnessWaybackMachine/site/index.html


Cell 4 — Ensure the remote is tokenized (only if GH_TOKEN provided)

In [4]:
# 4) If you provided GH_TOKEN above, temporarily set a tokenized remote for this session
import subprocess, shlex

def run(cmd):
    print("$", cmd)
    return subprocess.call(cmd, shell=True)

run("git remote -v")

if GH_TOKEN:
    url = f"https://{GITHUB_USERNAME}:{GH_TOKEN}@github.com/{GITHUB_USERNAME}/WorstellWitnessWaybackMachine.git"
    run("git remote remove {r} 2>/dev/null || true".format(r=REMOTE_NAME))
    run(f"git remote add {REMOTE_NAME} {shlex.quote(url)}")
    run("git remote -v")
else:
    print("GH_TOKEN empty → using whatever remote is already configured.")


$ git remote -v
$ git remote remove origin 2>/dev/null || true
$ git remote add origin https://BillWorstell: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@github.com/BillWorstell/WorstellWitnessWaybackMachine.git
$ git remote -v


Cell 5 — Publish with gh-deploy

In [5]:
# 5) Publish to GitHub Pages (creates/updates the gh-pages branch on the remote)
%cd $PROJECT_DIR
!mkdocs gh-deploy --force --remote-name $REMOTE_NAME --message "Publish from Colab"

/content/drive/MyDrive/WorstellWitnessWaybackMachine
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /content/drive/MyDrive/WorstellWitnessWaybackMachine/site
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - decades/1640s.md
  - downloads/Readme.md
  - families/volkert-dircks.md
INFO    -  Doc file 'places/smits-vly.md' contains an absolute link '/families/volckertszen-dircks/', it was left as is. Did you mean '../families/volckertszen-dircks.md'?
INFO    -  Doc file 'places/smits-vly.md' contains an absolute link '/decades/1600s/', it was left as is.
INFO    -  Doc file 'places/smits-vly.md' contains an absolute link '/decades/1640-1650/', it was left as is. Did you mean '../decades/1640-1650.md'?
INFO    -  Documentation built in 5.12 seconds
INFO    -  Copying '/content/drive/MyDrive/WorstellWitnessWaybackMachine/site' to 'gh-pages' branch and pushing to GitHub.
Enumerating objects: 13

Cell 6 — Confirm gh-pages exists on remote

In [6]:
# 6) Verify gh-pages branch was pushed
%cd $PROJECT_DIR
!git ls-remote --heads $REMOTE_NAME gh-pages

/content/drive/MyDrive/WorstellWitnessWaybackMachine
cbc7006377c34dd85e414e4649cba54ea83a6f82	refs/heads/gh-pages


Cell 7 — Check the live site (poll a few times)

In [7]:
# 7) Ping the live URL until it returns 200 (first publish can take a short while)
import time, requests

url = SITE_URL
ok = False
for i in range(10):   # up to ~2 minutes (10 * 12s)
    try:
        r = requests.get(url, timeout=15)
        print(f"Attempt {i+1}: HTTP {r.status_code}")
        if r.ok:
            ok = True
            break
    except Exception as e:
        print(f"Attempt {i+1}: error → {e}")
    time.sleep(12)

print("✅ Live!" if ok else "⏳ Not live yet — open repo Settings → Pages and confirm 'Deploy from branch: gh-pages / root'.")

Attempt 1: HTTP 404
Attempt 2: HTTP 404
Attempt 3: HTTP 404
Attempt 4: HTTP 200
✅ Live!


(Optional) Cell 8 — Switch your remote back to a clean URL

In [8]:
# 8) (Optional) Restore a non-token remote when you’re done publishing
if GH_TOKEN:
    !git remote set-url $REMOTE_NAME https://github.com/$GITHUB_USERNAME/WorstellWitnessWaybackMachine.git
!git remote -v

origin	https://github.com/BillWorstell/WorstellWitnessWaybackMachine.git (fetch)
origin	https://github.com/BillWorstell/WorstellWitnessWaybackMachine.git (push)


If you still see 404 after publishing

In GitHub: Repo → Settings → Pages

Build and deployment → Source: Deploy from a branch

Branch: gh-pages / Folder: / (root)

Reload your SITE_URL after ~30–60s.